File: 26099/frontend/capture2.mjs
import { chromium } from 'playwright'
import path from 'path'
import fs from 'fs'
import { fileURLToPath } from 'url'

const __dirname = path.dirname(fileURLToPath(import.meta.url))
const BASE = 'http://localhost:5173'
const OUT = path.join(__dirname, 'review')

// Actual routes from App.tsx, in demo flow order
const ROUTES = [
  { href: '/',            file: '02-dashboard.png' },
  { href: '/overview',    file: '03-overview.png' },
  { href: '/explorer',   file: '04-explorer.png' },
  { href: '/matching',    file: '05-matching.png' },
  { href: '/duplicates',  file: '06-duplicates.png' },
  { href: '/savings',     file: '07-savings.png' },
  { href: '/registry',    file: '08-registry.png' },
  { href: '/migration',   file: '09-migration.png' },
  { href: '/integration', file: '10-integration.png' },
  { href: '/import',      file: '11-import.png' },
  { href: '/normalize',   file: '12-normalize.png' },
  { href: '/activity',    file: '13-activity.png' },
  { href: '/engine',      file: '14-engine.png' },
]

;(async () => {
  const browser = await chromium.launch()

  // Login: fresh session (no sessionStorage)
  const loginCtx = await browser.newContext({ viewport: { width: 1400, height: 900 } })
  const loginPage = await loginCtx.newPage()
  const loginPath = path.join(OUT, '01-login.png')
  console.log('Capturing login ->', loginPath)
  await loginPage.goto(BASE + '/login', { waitUntil: 'networkidle', timeout: 30000 })
  await loginPage.waitForTimeout(500)
  await loginPage.screenshot({ path: loginPath })
  console.log('login:', fs.existsSync(loginPath) ? fs.statSync(loginPath).size + ' bytes' : 'MISSING')
  await loginCtx.close()

  // Authenticated pages: pre-seeded sessionStorage (3 CPSEs loaded)
  const ctx = await browser.newContext({ viewport: { width: 1400, height: 900 } })
  const page = await ctx.newPage()
  await page.addInitScript(() => {
    try {
      sessionStorage.setItem('codeone.role', 'reviewer')
      sessionStorage.setItem('codeone.name', 'Reviewer')
      sessionStorage.setItem('codeone.learned', '[]')
      sessionStorage.setItem('codeone.loaded', JSON.stringify(['IOCL', 'NTPC', 'SAIL']))
    } catch (e) { /* storage disabled */ }
  })

  for (const route of ROUTES) {
    const fp = path.join(OUT, route.file)
    console.log('Capturing', route.href, '->', route.file)
    await page.goto(BASE + route.href, { waitUntil: 'networkidle', timeout: 30000 })
    await page.waitForTimeout(800)
    await page.screenshot({ path: fp })
    const ok = fs.existsSync(fp)
    const sz = ok ? fs.statSync(fp).size : 0
    console.log('  written:', ok, 'size:', sz)
  }

  await browser.close()
  console.log('\nDone. All files:')
  for (const f of fs.readdirSync(OUT).filter(f => f.endsWith('.png')).sort()) {
    console.log(' ', f, fs.statSync(path.join(OUT, f)).size, 'bytes')
  }
})()

File: 26099/frontend/capture2.py
import os, time, json
from playwright.sync_api import sync_playwright

PORT = 5174
BASE_URL = f"http://localhost:{PORT}"
REVIEW_DIR = r"C:\Users\udayp\Documents\code\SIH26\26099\frontend\review"

PAGES = [
    ("/", "Dashboard"),
    ("/overview", "Overview"),
    ("/explorer", "Explorer"),
    ("/matching", "Matching"),
    ("/duplicates", "Duplicates"),
    ("/savings", "Savings"),
    ("/registry", "Registry"),
    ("/migration", "Migration"),
    ("/integration", "Integration"),
    ("/import", "Import"),
    ("/normalize", "Normalize"),
    ("/activity", "Activity"),
    ("/engine", "Engine"),
]

os.makedirs(REVIEW_DIR, exist_ok=True)

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page(viewport={"width": 1440, "height": 900})
    page.goto(f"{BASE_URL}/", wait_until="networkidle", timeout=30000)
    page.wait_for_timeout

    results = {}

    for path, label in PAGES:
        try:
            print(f"\n=== {label} ({path}) ===")
            page.goto(f"{BASE_URL}{path}", wait_until="networkidle", timeout=30000)
            page.wait_for_timeout

            safe = path.replace("/", "_").strip("_") or "home"
            out = os.path.join(REVIEW_DIR, f"{safe}.png")
            page.screenshot(path=out, full_page=False)
            print(f"  Screenshot: {out}")

            body_text = page.inner_text("main, [role=main], .main, main")
            # Get sidebar links to identify nav items visible
            sidebar = page.inner_text("aside, nav, .sidebar, [role=navigation]")

            # Get interactive elements
            buttons = page.eval_on_selector_all("button, [role=button]", "els => els.map(e => e.textContent?.trim()).filter(Boolean).slice(0,30)")
            links = page.eval_on_selector_all("a", "els => els.map(e => ({text: e.textContent?.trim().slice(0,60), href: e.getAttribute('href')})).filter(e => e.href).slice(0,30)")
            inputs = page.eval_on_selector_all("input, select, textarea", "els => els.map(e => ({tag: e.tagName, type: e.getAttribute('type'), placeholder: e.getAttribute('placeholder')})).slice(0,20)")

            results[label] = {
                "path": path,
                "body_text": body_text[:2000],
                "sidebar": sidebar[:500] if sidebar else "",
                "buttons": buttons,
                "links": links,
                "inputs": inputs,
            }
            print(f"  Text preview: {body_text[:300]}")
        except Exception as e:
            print(f"  FAILED: {e}")

    with open(os.path.join(REVIEW_DIR, "_page_data.json"), "w", encoding="utf-8") as f:
        json.dump(results, f, indent=2, ensure_ascii=False)

    browser.close()

print("\nDone!")
